lib/scala/pkg/src/main/scala/org/enso/pkg/ComponentGroup.scala

```diff
@@ -0,0 +1,316 @@
+package org.enso.pkg
+
+import io.circe._
+import io.circe.syntax._
+
+/** The description of component groups provided by the package.
+  *
+  * @param newGroups the list of component groups provided by the package
+  * @param extendedGroups the list of component groups that this package extends
+  */
+case class ComponentGroups(
+  newGroups: List[ComponentGroup],
+  extendedGroups: List[ExtendedComponentGroup]
+)
+object ComponentGroups {
+
+  /** An empty component groups. */
+  val empty: ComponentGroups =
+    ComponentGroups(List(), List())
+
+  /** Fields for use when serializing the [[ComponentGroups]]. */
+  private object Fields {
+    val New     = "new"
+    val Extends = "extends"
+  }
+
+  /** Does the provided JSON object have unknown fields. */
+  private def hasUnknownKeys(cursor: HCursor): Boolean =
+    (ConfigCodecs.objectKeys(cursor) - Fields.New - Fields.Extends).nonEmpty
+
+  /** [[Encoder]] instance for the [[ComponentGroups]]. */
+  implicit val encoder: Encoder[ComponentGroups] = { componentGroups =>
+    val newGroups = Option.unless(componentGroups.newGroups.isEmpty)(
+      Fields.New -> componentGroups.newGroups.asJson
+    )
```

Comment:
Wow, why I didn't know that you can use `unless` this way - awesome! :tada: 
It would have made quite a lot of other serialization code that I was writing much cleaner.